src/runtime/types.cpp

```diff
@@ -3728,11 +3728,12 @@ void setupRuntime() {
         BoxedClass(object_cls, BoxedCApiFunction::gcHandler, 0, 0, sizeof(BoxedCApiFunction), false, "capifunc");
     method_cls = new (0) BoxedClass(object_cls, BoxedMethodDescriptor::gcHandler, 0, 0, sizeof(BoxedMethodDescriptor),
                                     false, "method_descriptor");
+    method_cls->giveAttr("__name__", boxString("method_descriptor"));
```

Comment:
I don't think this is quite right -- I think this is supposed to be the name of the method object itself, not the class.  We should add `print object.__reduce__.__name__` to the test -- CPython says `__reduce__` and we will say `method_descriptor`.  I think this should be something like 
```C++
method_cls->giveAttr("__name__", new BoxedMemberDescriptor(BoxedMemberDescriptor::OBJECT,
                                                       offsetof(BoxedMemberDescriptor, name));
```
ie this is the magic to say that the `__name__` attribute should proxy to the C field `BoxedMemberDescriptor::name`.  This won't work right away since BoxedMemberDescriptor doesn't have a `name` field, so one would need to be added.